checker/src/main/java/org/checkerframework/checker/signedness/SignednessAnnotatedTypeFactory.java

```diff
@@ -115,12 +115,12 @@ private void addSignednessGlbAnnotation(Tree tree, AnnotatedTypeMirror type) {
                         ExpressionTree expr = t.getExpression();
                         AnnotatedTypeMirror varType = getAnnotatedType(var);
                         AnnotatedTypeMirror exprType = getAnnotatedType(expr);
-                        AnnotatedTypeMirror lht = getAnnotatedType(var);
                         if (!(varType.hasAnnotation(Unsigned.class)
                                         && exprType.hasAnnotation(Signed.class))
                                 && !(varType.hasAnnotation(Signed.class)
-                                        && exprType.hasAnnotation(Unsigned.class)))
-                            type.replaceAnnotations(lht.getAnnotations());
+                                        && exprType.hasAnnotation(Unsigned.class))
+                                && !varType.hasExplicitAnnotation(SignednessGlb.class))
```

Comment:
If the variable is annotated explicitly with `@SigndnessGlb` then the type annotation should not get replaced.

Reply:
> Please use descriptive commit messages. "Update file" does not describe the change, so it is not helpful in understanding what you have done.

I have explained the work of that commit in this comment.